core-rust/state-manager/src/transaction/round_update_transaction.rs

```diff
@@ -152,6 +152,11 @@ impl PreparedRoundUpdateTransactionV1 {
                 },
                 pre_allocated_addresses: vec![],
             },
+            // Note: technically the round update is a "system transaction" (which would justify
```

Comment:
Hmm it should be true I think... any existing limits should never be hit and can effectively be 
 reasoned as being infinite.
Round changes can be quite large if they're an epoch change... So I think we should set this as True and potentially get rid of the `ConfigType::OtherSystem` type to simplify things.

Paging @talekhinezh for advice/input too.

Reply:
After the other changes (i.e. from other review comments) here, my `// Note:` is no longer true (i.e. we do not customize `LimitParameters` now).
I also consider the "round update" a system transaction (and I was surprised we wanted to e.g. track limits for it before - but maybe someone had a reason?).

> get rid of the `ConfigType::OtherSystem` type to simplify things

do you suggest to treat it as `::Regular`? (plz note the same would have to happen for `FlashV1`)
Logic-wise, it seems that would be totally equivalent; but maybe conceptually we want to have this special label anyway? (I don't have very strong opinion)

Reply:
BTW: `ConfigType::Pending` is equivalent to `ConfigType::Regular` too, at this point. I can accept getting rid of it.

Reply:
Conclusion:
- let's get rid of `ConfigType::OtherSystem`, because `system: true` does all this anyways.
- Engine can (and probably should?) get rid of `ExecutionConfig::for_system_transaction` & co - FYI @talekhinezh 
- the `ConfigType::Pending` stays, because we use it to differentiate some log warn threshold.